code/modules/recycling/wrapped_package.dm

```diff
@@ -0,0 +1,283 @@
+/////////////////////////////////////////////////////////////////////
+// Parcel
+/////////////////////////////////////////////////////////////////////
+
+/**
+ * A parcel wrapper for items and structures that can be wrapped in wrapping paper.
+ */
+/obj/item/parcel
+	name              = "parcel"
+	desc              = "A wrapped package."
+	icon              = 'icons/obj/items/storage/deliverypackage.dmi'
+	icon_state        = "parcel"
+	item_flags        = ITEM_FLAG_HOLLOW
+	material          = /decl/material/solid/paper
+	attack_verb       = list("delivered a hit", "expedited on", "shipped at", "went postal on")
+	base_parry_chance = 40 //Boxes tend to be good at parrying
+	///A text note attached to the parcel that shows on examine
+	var/attached_note
+	///The first part of the icon_state names
+	var/tmp/icon_state_prefix = "parcel"
+	///Color of the dropped paper trash on unwrapping.
+	var/tmp/trash_color = "#b97644"
```

Comment:
Tmp does nothing besides making the proc returning whether a var is saved or not return false. 
tmp is for things that shouldn't be saved or are just temporary. Like constants and the like. Since constants cannot be changed by subclass, might as well mark them tmp, since they never change from their definition and will never be saved.

Reply:
The problem here, is that constants cannot be changed in sub-classes. 
So nobody uses the const keyword for that, unless the var is also static.

Reply:
I would just let them be normal vars, I feel that's the most descriptive and idiomatic approach. I don't care strongly if you insist on tmp, I guess.

Reply:
> I would just let them be normal vars, I feel that's the most descriptive and idiomatic approach. I don't care strongly if you insist on tmp, I guess.

That's a bit of an oxymoron, no? Why would not marking a var as unsaved would be more idiomatic and descriptive than not? 
For the idiomatic part, that's literally how the language wants you to use it. So you can hardly be more idiomatic, no? 
Its also descriptive, in indicating that it shouldn't be saved? Which implies that its either temporary or never changes from definition?


> If you're changing the value, it's not a constant. :/

I meant constant as in it doesn't change for the lifetime of the object. Its defined once and doesn't change.
Those probably would have been better as procs I guess. But, then there's the overhead. So I dunno.

Reply:
Idiomatic means using it the way it's used in the codebase or broader ss13 context; it's not used like this in the codebase. Idioms are about active usage, not some kind of language-design concepts. It's also not descriptive, as it's not temporary in any meaningful sense--it's the opposite, you are trying to indicate that it's immutable.

I agree it can't be made const. You can wrap it in a proc if you want but that's way afield from why I commented on this in review.

Reply:
I mean, its not like ss13 codebases will go out of their way to support a feature of DM they never use. 
Also, I can't help it that this is what they picked as name for their variable qualifier. It literally just means "not a constant, not a system variable, but still unsaved". Its not too far fetched to think someone that knows the language would be expected to know what that's for considering its in the doc?

Reply:
I looked up `tmp` in the dm documentation. Here is what it says:
```
The tmp type modifier indicates that an object variable should not be automatically written to the save file. This could mean that the variable is transient—that is, it is calculated at run-time and need not be saved. It could also indicated that the designer will handle saving of that variable specially and wishes to bypass the automated routine.
```
I don't buy your basic premise that this paragraph suggests that one should use `tmp` to designate immutable but type-specific vars. A literal reading would indicate that it should be used for any state var holding a reference, as none of those should be "written to a save file" in any reasonable context (indeed, the--admittedly ridiculous--examples following this in the dm docs make this very point). Along similar chains of reasoning, other vars that should have tmp are vars computed at runtime, like in `Initialize` or `Process`, and vars holding privileged information. But state vars pointing to static types are safe to store in savefiles, and so I see no reason why they would receive the `tmp` designation. You could argue that they are useless to store in savefiles if they are immutable (fine, sure)--but then you have to follow the actual logic and use `tmp` in all of the cases I just described, which will apply to most instance vars in the codebase.

This is of course all moot, because _we don't use savefiles_.

As I said, if you insist on `tmp` here, fine. But don't tell me your usage is specifically in line with the language architecture's intent, because that's not any more or less true than with any other sane usage pattern, such as e.g. never using it.